kafka/producer.go

```diff
@@ -287,10 +287,9 @@ func channelProducer(p *Producer) {
 
 // channelBatchProducer serves the ProduceChannel channel and attempts to
 // improve cgo performance by using the produceBatch() interface.
-func channelBatchProducer(p *Producer) {
+func channelBatchProducer(p *Producer, batchSize int) {
```

Comment:
I dont believe there is any use for synchronizing the batch size to the channel size:
since the channel is read message-by-message it is possible (likely and desired) to have a batch larger than the channel size.

The experimental batch interface aims to cut down on the number of C calls (since cgo has a high overhead), there is thus no logical correlation to the channel size.